runtime/src/chpl-file-utils.c

```diff
@@ -204,3 +203,58 @@ qioerr chpl_fs_remove(const char* name) {
     err = qio_mkerror_errno();
   return err;
 }
+
+qioerr chpl_fs_samefile(int* ret, qio_file_t* file1, qio_file_t* file2) {
+  qioerr err = 0;
+  struct stat f1;
+  struct stat f2;
+
+  int exitStatus = fstat(file1->fd, &f1);
+  if (exitStatus) {
+    // An error occurred.  Return it.
+    err = qio_mkerror_errno();
+    return err;
+  }
+  exitStatus = fstat(file2->fd, &f2);
+  if (exitStatus) {
+    // An error occurred.  Return it.
+    err = qio_mkerror_errno();
+  } else {
+    if (f1.st_dev == f2.st_dev && f1.st_ino == f2.st_ino) {
```

Comment:
Is there a plan to expose stat() as a library function in the future (I didn't see it in the current proposal)? If so, maybe note in samefile() that it can/should use stat() when it becomes available.